### PR TITLE
Set ml-commons plugin 3.0.0 baseline JDK version to JDK-21

### PR DESCRIPTION
**Pull request by reta**
### Description
Set ml-commons plugin 3.0.0 baseline JDK version to JDK-21
 
### Issues Resolved
Closes https://github.com/opensearch-project/ml-commons/issues/2503
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: client/build.gradle [hunk 1]
```diff
@@ -7,9 +7,9 @@ plugins {
     id 'java'
     id "io.freefair.lombok"
     id 'jacoco'
-    id 'com.github.johnrengelman.shadow'
+    id 'io.github.goooler.shadow' version "8.1.7"
```
Comment (reta) [inline client/build.gradle:10]: See please https://github.com/johnrengelman/shadow/issues/908

File: build.gradle [hunk 2]
```diff
@@ -43,6 +44,16 @@ buildscript {
             }
         }
     }
+
+    configurations {
+        classpath {
+            resolutionStrategy {
```
Comment (reta) [inline build.gradle:50]: See please https://github.com/johnrengelman/shadow/pull/876#issuecomment-1942380071

**Comment by dhrubo-os** [comment]
```
 > Task :opensearch-ml-common:compileJava FAILED

> Task :opensearch-ml-client:generateTestEffectiveLombokConfig
FAILURE: Build failed with an exception.
> Task :opensearch-ml-client:generateEffectiveLombokConfig
> Task :opensearch-ml-memory:generateEffectiveLombokConfig


Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

* What went wrong:
You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.
Execution failed for task ':opensearch-ml-common:compileJava'.

> error: invalid source release: 21
For more on this, please refer to https://docs.gradle.org/8.8/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
```

**Comment by reta** [comment]
@dhrubo-os thanks for looking, we need https://github.com/opensearch-project/ml-commons/pull/2584 first

**Comment by reta** [comment]
@dhrubo-os could you please recheck this one? thank you

File: .github/workflows/CI-workflow.yml [hunk 3]
```diff
@@ -101,6 +101,11 @@ jobs:
           role-to-assume: ${{ secrets.ML_ROLE }}
           aws-region: us-west-2
 
+      - name: Setup Java ${{ matrix.java }}
```
Comment (reta) [inline .github/workflows/CI-workflow.yml:104]: It should fix the compilation problem (but sadly GA does not pick the workflow from the pull request)

**Review by dhrubo-os — APPROVED** [review]

**Comment by reta** [comment]
@dhrubo-os anything holding off the merge? (the GA worflow should be fixed by this change) thank you!

**Review by mingshl — APPROVED** [review]

**Pull request merged**